unicodetools/src/main/java/org/unicode/text/UCD/GenerateData.java

```diff
@@ -811,6 +814,57 @@ public static void writeNormalizerTestSuite(String directory, String fileName)
         for (final String testSuiteCase : testSuiteCases) {
             writeLine(testSuiteCase, log, false);
         }
+        // At least one implementation (ICU4X) has an edge case when a character
+        // whose decomposition contains multiple starters and ends with a
+        // non-starter is followed by a non-starter of lower CCC.
+        // See https://github.com/unicode-org/unicodetools/issues/656
+        // and https://github.com/unicode-org/icu4x/pull/4530.
+        // That implementation also has separate code paths for the BMP and
+        // higher planes.  No such decompositions currently exist outside the
+        // BMP, but by generating these test cases we ensure that this would be
+        // covered.
+        // We stick them in Part 0, which is in principle for handcrafted test
+        // cases, because there are not many of them, and the edge case feels a
+        // tad too weird to describe in the title of a new part.
+        final org.unicode.props.UnicodeProperty sc =
+                IndexUnicodeProperties.make().getProperty(UcdProperty.Script);
+        final org.unicode.props.UnicodeProperty ccc =
+                IndexUnicodeProperties.make().getProperty(UcdProperty.Canonical_Combining_Class);
+        loopOverCodePoints:
+        for (final String cp : UnicodeSet.ALL_CODE_POINTS) {
+            final String[] decompositions =
+                    new String[] {Default.nfd().normalize(cp), Default.nfkd().normalize(cp)};
```

Comment:
Nit: Should we cache things like Default.nfd(), Default.nfkd(), Default.ucd(), IUP.make(), ... in variables in this function, or in instance fields? That should be easy, might save a bit of time on class synchronization, and make these call sites shorter (more readable). WDYT?

---
I *do not* like to cache this kind of stuff in *static* class fields, because if something goes wrong in the static initializer it's cumbersome to debug.

Reply:
> WDYT?

Yes, but in a subsequent cleanup PR, because a cursory search for .nfd() finds 18 results of which 7 are in commented-out code, and with the file in that state I do not want to waste further time figuring out which ones are in the same function; nor do I want to burn the commented-out code to the ground in this unrelated change.